### DISCUSSION TITLE
Building iOS .ipa from command line

### DISCUSSION THREAD

**Discussion by baskren** [unoplatform/uno#17841] [General]
First, [the official Microsoft document](https://learn.microsoft.com/en-us/dotnet/maui/ios/deployment/publish-cli?view=net-maui-8.0) does not work.  Hasn't in over a year.  My work around for this was pretty awful, required VS Mac, and can't be automated.  

However, there is a way to get an `.ipa` from the command line, as I discovered [here](https://github.com/xamarin/xamarin-macios/issues/20958).

```command-line
MyUnoApp> dotnet build -c  Release -f net8.0-ios -r ios-arm64 -t:publish
```

The above assumes the following:

1) The command is run in the same folder as your Uno Platform application project
2) There's only one `.csproj` in your Uno Platform application project folder
3) Your app's `CodesignKey` and optionally its `CodesignProvision` are set as properties in its `.csproj`

**Comment by francoistanguay** [comment]
Is this doc helpful as well? https://platform.uno/docs/articles/uno-publishing-ios.html

**Reply by baskren** [reply]
VERY!
